subsys/bluetooth/audio/codec.c

```diff
@@ -123,13 +123,16 @@ int bt_codec_cfg_get_chan_allocation_val(const struct bt_codec *codec,
 
 	*chan_allocation = 0;
 	if (bt_codec_get_val(codec, BT_CODEC_CONFIG_LC3_CHAN_ALLOC, &element)) {
-
 		*chan_allocation = sys_le32_to_cpu(*((uint32_t *)&element->data.data[0]));
 
 		return BT_AUDIO_CODEC_PARSE_ERR_SUCCESS;
 	}
 
-	return BT_AUDIO_CODEC_PARSE_ERR_TYPE_NOT_FOUND;
+	/* BAP v1.0.1 4.3.2 Codec_Specific_Configuration LTV requirements
+	 * The absence of the Audio_Channel_Allocation LTV structure
+	 * shall be interpreted as a single channel with no specified Audio Location.
+	 */
+	return BT_AUDIO_LOCATION_UNSPECIFIED;
```

Comment:
What is the reasoning here to return `0` (`BT_AUDIO_LOCATION_UNSPECIFIED`, which is btw also an `bt_audio_location` and not an `int`) here, instead of an error code? 

Seems odd to return error code in all other instances, except this one (and even more weirdly to return 0, i.e. success, in that case). 

If we just want to return 0/sucess, instead of an error, I suggest that we just return 0 directly, instead of putting `BT_AUDIO_LOCATION_UNSPECIFIED` into `bt_audio_location` and use that here.